### PR TITLE
Implement RTCPeerConnection events

### PR DESCRIPTION
**Pull request by rogurotus**
## Synopsis

Implement RTCPeerConnection events in PeerConnectionObserver callbacks.

## Checklist

- Created PR:
    - [x] In [draft mode][l:1]
    - [x] Name contains `Draft: ` prefix
    - [x] Name contains issue reference
    - [x] Has `k::` labels applied
    - [x] Has assignee
- [x] Documentation is updated (if required)
- [x] Tests are updated (if required)
- [x] Changes conform code style
- [x] CHANGELOG entry is added (if required)
- [x] FCM (final commit message) is posted
    - [x] and approved
- [x] [Review][l:2] is completed and changes are approved
- Before merge:
    - [x] Milestone is set
    - [x] PR's name and description are correct and up-to-date
    - [x] `Draft: ` prefix is removed
    - [x] All temporary labels are removed

[l:1]: https://help.github.com/en/articles/about-pull-requests#draft-pull-requests
[l:2]: https://help.github.com/en/articles/reviewing-changes-in-pull-requests

### CODE REVIEW

**Comment by rogurotus** [comment]
## FCM
```
Implement `RTCPeerConnection events` (#27)

Additionally:
- `OnSignalingChange` event
- `OnStandardizedIceConnectionChange` event
- `OnConnectionChange` event
- `OnIceGatheringChange` event
- `OnNegotiationNeededEvent` event
- `OnIceCandidateError` event
- `OnIceCandidateError` event (args overload)
- `OnIceConnectionReceivingChange` event
- `OnInterestingUsage` event
- `OnIceCandidate` event
- `OnIceCandidatesRemoved` event
- `OnIceSelectedCandidatePairChanged` event
```

**Review by alexlapa — APPROVED** [review]

File: crates/libwebrtc-sys/src/bridge.rs [hunk 1]
```diff
@@ -71,6 +81,148 @@ pub(crate) mod webrtc {
...
+        kIceConnectionDisconnected,
+
+        /// [RTCIceConnectionState.closed][1] representation.
+        ///
+        /// [1]: https://w3.org/TR/webrtc#dom-rtciceconnectionstate-closed
+        kIceConnectionClosed,
+
+        /// Non-spec-compliant variant. `WebRTC` docs state that it is
+        /// unreachable.
+        kIceConnectionMax,
```
Comment (tyranron) [inline crates/libwebrtc-sys/src/bridge.rs:223]: @rogurotus can you provide a link to that?

File: crates/libwebrtc-sys/src/bridge.rs [hunk 2]
```diff
@@ -469,6 +686,69 @@ pub(crate) mod webrtc {
...
+        );
+
+        /// Completes the provided [`DynPeerConnectionEventsHandler`].
+        pub fn on_standardized_ice_connection_change(
+            cb: &mut DynPeerConnectionEventsHandler,
+            new_state: IceConnectionState,
+        );
+
+        /// Completes the provided [`DynPeerConnectionEventsHandler`].
+        pub fn on_connection_change(
```
Comment (tyranron) [inline crates/libwebrtc-sys/src/bridge.rs:702]: @rogurotus here all the docs are same, which isn't quite useful. You should describe for the each function in what circumstances it completes the handler.

File: windows/src/peer_connection.cc [hunk 3]
```diff
@@ -47,6 +51,120 @@ class SetDescriptionCallBack : public SetDescriptionCallbackInterface {
...
+    if (deps_->sink_) {
+      flutter::EncodableMap params;
+      params[EncodableValue("event")] = "OnIceConnectionStateChange";
+      params[flutter::EncodableValue("state")] = new_state;
+      deps_->sink_->Success(flutter::EncodableValue(params));
+    }
+  };
+
+  // Propagates an event to the Flutter side via `EventSink`.
+  void OnIceGatheringStateChange(const std::string& new_state) {
```
Comment (tyranron) [inline windows/src/peer_connection.cc:132]: Similarly here, better describe in docs which event is propagated exactly.

**Review by tyranron — CHANGES_REQUESTED** [review]

**Review by tyranron — APPROVED** [review]

**Pull request merged**